index/scorch/persister.go

```diff
@@ -113,13 +114,27 @@ OUTER:
 		case s.introducerNotifier <- w:
 		}
 
-		s.removeOldData() // might as well cleanup while waiting
+		s.removeOldData() // this looks enough as of now
```

Comment:
i understood the old comment, but i don't understand the new one

Reply:
hm, I didn't fully understood the original comment. :) As there is no waiting there right...we just happen to do this in between the persister notification sending and receiving the response from it..Those two could just happen within a speck time and we are not doing anything in while wait loop, isn't it. 
And made the comment this way, in context of our the file count handling document. A thread there talks about 

while(condition) {

s.removeOldData() 
until few of the break conditions met...
}
So as of now, it looks like we could manage with a single call than in a loop until certain conditions are met.

Reply:
the "waiting" refers to the fact that we just notified the introducer that we're all caught up to a certain point, and we're about to hit a select statement that will "wait" for notification of more work to be done.  the duration of the wait may be 0 though.

i don't understand the while loop you posted here, if this actual code could you link to it?

finally, the words "this looks enough as of now" doesn't make any sense to me, like if you walked up to me on the street, pointed at something and said those words, i would have no idea...